libs/ojsonnet/Eval_jsonnet.ml

```diff
@@ -190,7 +190,16 @@ and eval_expr (env : V.env) (v : expr) : V.value_ =
                    fst field.fld_name = fld)
           with
           | None -> error tk (spf "field '%s' not present in %s" fld (sv e))
-          | Some fld -> evaluate_lazy_value_ fld.fld_value)
+          | Some fld -> (
+              let new_self = V.Object (_l, (_assertsTODO, fields), _r) in
```

Comment:
maybe add the comment in the PR here also.

Reply:
Also for people not familiar with the jsonnet spec, it might be useful to give a quick summary of the semantic of self/super here (ok maybe not super, but at least self).

Reply:
I have a hard time for example understand why this code works better than what was before where we were also assigning LSelf to new_self.
Do you have an example showing why we need the code the way it's written and why before where we bind LSelf in
eval_obj_inside it could not work?

Reply:
Then we could put this example in comment and show the trace of its evaluation and expose the subtelities.

Reply:
Yep, can add. Here's a small specific example is the following (will also add it to code). 
`({ name : self.y } + {y : 3})["name"]`
If we do the binding to self in eval_obj_inside, we end up with self bound to 'o = {name:self.y}'. Then evaluating the plus would give (semantically)
'({name : o.y, y : 3}) 'and when doing the access, we try to evaluate  {name:self.y}.y which doesn't exist. However, if we delay self/super binding til access (like in the spec) we end up with
'({name : self.y, y : 3})["name"]' then on access, self gets bound to 'o = {name : self.y, y:3}' 
so we get o.y when we access name, and now are left with the correct return (3).

Reply:
great example!